build.sbt

```diff
@@ -32,13 +34,16 @@ assemblyJarName in assembly := s"${name.value}-uber.jar"
 
 assemblyMergeStrategy in assembly := {
   case "META-INF/io.netty.versions.properties" => MergeStrategy.last
+  // engine uses bblfsh client scala, which also uses scalapb but different version
+  case PathList("scalapb", xs @ _*) => MergeStrategy.first
+  case PathList("com", "trueaccord", "lenses", xs @ _*) => MergeStrategy.first
+  case PathList("com", "google", "protobuf", xs @ _*) => MergeStrategy.first
   case x =>
     val oldStrategy = (assemblyMergeStrategy in assembly).value
     oldStrategy(x)
 }
 
 assemblyShadeRules in assembly := Seq(
-  ShadeRule.rename("com.google.common.**" -> "com.google.shadedcommon.@1").inAll,
```

Comment:
I strongly suggests we keep this shading, otherwise chances are very high we'll have problems in Spark runtime at cluster mode.

Reply:
It breaks build.
- `./sbt package`
- `./sbt assembly`
- `./sbt package` - fails with error expected `com.google.common` got `com.google.shadedcommon`

let me know how to avoid it without removing shading

Reply:
Let's keep it like that, after it's proven with https://github.com/src-d/gemini/pull/63#issuecomment-374553457 not to cause us troubles with modern version of engine on the cluster mode, after an awesome work done by @carlosms .

As soon as #72 is merged, I guess a rebase is in order.